lib/raygun.ts

```diff
@@ -240,26 +211,75 @@ class Raygun {
 
     if (!sendOptionsResult.valid) {
       console.error(
-        `[Raygun4Node] Encountered an error sending an error to Raygun. No API key is configured, please ensure .init() is called with api key. See docs for more info.`,
+        `[Raygun4Node] Encountered an error sending an error to Raygun. No API key is configured, please ensure .init is called with api key. See docs for more info.`,
       );
-      return sendOptionsResult.message;
+      return Promise.reject(sendOptionsResult.message);
     }
 
     const sendOptions = sendOptionsResult.options;
-
     if (this._isOffline) {
-      // make the save callback type compatible with Callback<IncomingMessage>
-      const saveCallback = callback
-        ? (err: Error | null) => callVariadicCallback(callback, err, null)
-        : emptyCallback;
-      this.offlineStorage().save(JSON.stringify(message), saveCallback);
+      // Server is offline, store in Offline Storage
+      return new Promise((resolve, reject) => {
+        this.offlineStorage().save(JSON.stringify(message), (error) => {
+          if (error) {
+            console.error(
+              "[Raygun4Node] Error storing message while offline",
+              error,
+            );
+            reject(error);
+          } else {
+            debug("[raygun.ts] Stored message while offline");
+            // Resolved value is null because message is stored
+            resolve(null);
+          }
+        });
+      });
     } else {
+      // wrap Promise and add duration debug info
+      const stopTimer = startTimer();
       // Use current transport to send request.
       // Transport can be batch or default.
-      this.transport().send(sendOptions);
+      return this.transport()
+        .send(sendOptions)
+        .then((response) => {
+          const durationInMs = stopTimer();
+          debug(
+            `[raygun.ts] Successfully sent message (duration=${durationInMs}ms)`,
+          );
+          return response;
+        })
+        .catch((error) => {
+          const durationInMs = stopTimer();
+          debug(
+            `[raygun.ts] Error sending message (duration=${durationInMs}ms): ${error}`,
+          );
+          return error;
+        });
```

Comment:
Wrap the `transport.send` call with a `then/catch` so we can log our own error/success messages, but still send the results back to the end user.

This is what was done in `wrappedCallback`.